include/zephyr/stepper/step_dir_stepper_controller.h

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright 2024 Fabian Blatz <fabianblatz@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_STEP_DIR_STEPPER_CONTROLLER_H_
+#define ZEPHYR_INCLUDE_STEP_DIR_STEPPER_CONTROLLER_H_
```

Comment:
Does this header really need to be public? The functions are more or less the same as the ones in the API.

Reply:
True - I guess this can be moved to the stepper dir. Done.